google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/gcs/GcsBucketPanel.java

```diff
@@ -124,14 +115,15 @@ void refresh() {
 
     if (!Services.getLoginService().isLoggedIn()) {
       notificationLabel.setText(GctBundle.message("gcs.panel.bucket.listing.not.logged.in"));
-    } else if (StringUtils.isEmpty(projectSelector.getText())) {
+    } else if (projectSelector.getSelectedProject() == null) {
       notificationLabel.setText(GctBundle.message("gcs.panel.bucket.listing.no.project.selected"));
     } else {
-      String projectId = projectSelector.getText();
-      CredentialedUser user = projectSelector.getSelectedUser();
+      CloudProject cloudProject = projectSelector.getSelectedProject();
+      Optional<CredentialedUser> user =
+          Services.getLoginService().getLoggedInUser(cloudProject.googleUsername());
 
-      if (user != null) {
-        loadAndDisplayBuckets(projectId, user);
+      if (user.isPresent()) {
+        loadAndDisplayBuckets(cloudProject.projectId(), user.get());
```

Comment:
I really don't like called `get` on an `Optional`. But I suppose its better than this:

```
userOption.map(user -> loadAndDisplayBuckets(...)).orElseGet(notificationLabel.setText(..));
```

wdyt?

I wish java had better ways to handle this. I guess we have to wait until pattern matching makes it into the jdk.

Reply:
I would agree that multi-line `if` instead of one liner `ifPresent().orElse()` is easier to read.

Reply:
well it wouldn't necessarily be a one liner. the formatter would probably put it on multiple lines.

Either way, using map purely for a side effect is ugly too. So I guess would leave it.